### PR TITLE
zml: remove pjrtx

### PR DESCRIPTION
**Pull request by gwenzek**
pjrtx.zig was mean to be a ZML specific wrapper over pjrt.zig.
But pjrt.zig in its current form is pretty usable without wrapper,
the extra layer of indirection is IMHO confusing because it mostly transfer the calls directly to pjrt.zig but make some call suspending.

I moved the few helpers left in pjrtx.zig where they are used, ie 

`pjrtx.Buffer.bufferFromHostBuffer`-> in `buffer.Buffer.from`
`pjrtx.Client.compile` -> in `module.compileModuleToPjrtExecutable`
`pjrtx.Event.await` -> `zml.Platform.awaitEvent()`

This also fix a bug in `pjrtx.Event.await` which was blocking for the event before suspending for the event.

### CODE REVIEW

**Comment by steeve** [comment]
bold, I like it

File: zml/platform.zig [hunk 1]
```diff
@@ -94,7 +94,40 @@ pub const Platform = struct {
...
         return self.pjrt_client.getProfiler(self.pjrt_api, options);
     }
+
+    /// Suspend the current co-routine while awaiting for a pjrt event to be over.
+    pub fn awaitEvent(self: Platform, event: *pjrt.Event) !void {
+        defer event.deinit(self.pjrt_api);
+        // If we aren't in a coroutine just use the normal blocking api.
+        if (!asynk.inCoro()) {
+            return try event.await_(self.pjrt_api);
+        }
```
Comment (Corendos) [inline zml/platform.zig:107]: I get the motivation, but it feels like implementation details leaking.

I don't really have another suggestion currently though

File: zml/tensor.zig
Comment (Corendos) [inline zml/tensor.zig:1]: I'd argue that this should be in another PR because it's not related to removing `pjrtx.zig`

**Review by Corendos — APPROVED** [review]
Except the two comments I left, I like the fact that this PR removes more code than it adds 😁

File: zml/tensor.zig
Comment (gwenzek) [reply zml/tensor.zig:1]: I just stumbled by accident on it. But sure. Let me move that to another PR

File: zml/platform.zig (hunk 1 above)
Comment (gwenzek) [reply zml/platform.zig:107]: I could remove it, but I find it simpler to use the blocking api if asynk is not enabled.

File: zml/tensor.zig
Comment (gwenzek) [reply zml/tensor.zig:1]: actually I had reworked it because this disabled test is using "pjrt". But it shouldn't I just remove the test cause `test dynamicSlice` is already pretty similar.
